swupdate: make hawkbit server ip configurable

fahadarslan opened the pull request:
Signed-off-by: Fahad Arslan <Fahad_Arslan@mentor.com>

fahadarslan commented:
This is connected with
https://stash.alm.mentorg.com/projects/MEL/repos/mel-meta-mentor-swupdate/pull-requests/11/overview

@chombourger, @Noor-Ahsan kindly review!

chombourger commented on meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend:
```diff
@@ -22,6 +22,8 @@ do_install_append() {
     install -m 644 ${WORKDIR}/hwrevision ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+
+    sed -i "s/hawkbit/${HAWKBIT_SERVER_IP}/g" ${D}${systemd_unitdir}/system/swupdate.service
```
we may want to make the complete URL user-configurable

chombourger replied:
update the service file to have a clear pattern. e.g. @hawkbit_url@ and use sed to replace it with the HAWKBIT_SERVER_URL setting from local.conf (or its default value, which needs to be defined somewhere)

fahadarslan committed:
swupdate: derive swupdate.service from template

Provide template swupdate.service.in. Replace @hawkbit_url@
expression in template with user provided Hawkbit server URL in
local.conf and generate swupdate.service.

Signed-off-by: Fahad Arslan <Fahad_Arslan@mentor.com>

chombourger commented on meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend:
```diff
@@ -22,6 +23,9 @@ do_install_append() {
     install -m 644 ${WORKDIR}/hwrevision ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+
+    sed -e "s@\@hawkbit_url\@@${HAWKBIT_SERVER_URL}@g" ${WORKDIR}/swupdate.service.in > ${WORKDIR}/swupdate.service
```
would have use a different separator to avoid escaping (such as ,) but OK

chombourger commented:
looks good - thanks
